test/PlatesEngineFactoryTest.php

```diff
@@ -271,6 +271,41 @@ public function testExceptionIsRaisedIfMultiplePathsInSameNamespace(): void
         $factory($this->container->reveal());
     }
 
+    public function testSetExtensionByTemplatesConfig(): void
+    {
+        $config = [
+            'templates' => [
+                'extension' => 'html.twig',
+            ],
+        ];
+        $this->container->has('config')->willReturn(true);
+        $this->container->get('config')->willReturn($config);
+        $factory = new PlatesEngineFactory();
+
+        $engine = $factory($this->container->reveal());
+
+        $this->assertSame('html.twig', $engine->getFileExtension());
+    }
+
+    public function testOverrideExtensionByPlatesConfig(): void
+    {
+        $config = [
+            'templates' => [
+                'extension' => 'html.twig',
+            ],
+            'plates'    => [
+                'extension' => 'plates.php',
+            ],
```

Comment:
Ok, now that I can read the patch and understand it (thanks for the tests!), one question remains: why? :D 

Why is this necessary? I understand what you are doing, but is this just because `'templates'` and `'plates'` were (in theory) supposed to be one? :thinking:

Reply:
@Ocramius Twig and Plates are using `templates` config. 

If I have both installed https://github.com/mezzio/mezzio-twigrenderer and https://github.com/mezzio/mezzio-platesrenderer and are overwriting each other.

I would personally would say `templates` should be removed and `twig` should be configured via `twig` and `plates` should be configured via `plates` config option and not have a general `templates` option which can conflict with each other. Because if I configure a directory I already know if that directory contains twig files or will contain plates file, same for the (file)extension I know when I configure extension that the template files will be plates or twig so I can configure it for that specific package.

Reply:
`'templates'` is kinda the default when using `laminas/laminas-view`, so it won't go away anyway :)

Reply:
I see :) but maybe it is then only for the laminias-view configuration :)

Reply:
Absolutely, just saying that `'plates'` is secondary in the ecosystem.

Anyway, the use-case seems clear now.

@froschdesign you worked with plates for `laminas.dev`: WDYT? OK as-is?